### ISSUE TITLE
can't compile from source

### ISSUE THREAD

**Issue by ilteroi** [mikeboers/PyAV#97]
after my fail with the pip version of pyav, i tried to get the bleeding edge from git, but i cannot compile it:

Could not find avformat with ctypes; looked in:
	libavformat.so.54

but it's installed

$ dpkg --list | grep avformat
ii  libavformat-dev                             6:9.18-0ubuntu0.14.04.1                    amd64        Development files for libavformat
ii  libavformat54:amd64                         6:9.18-0ubuntu0.14.04.1                    amd64        Libav file format library

can you give me a hint how to fix this?

ps: i had to modify setup.py a little to even get this message, otherwise it crashed. line 70 should say: for root in extension_extra.get('library_dirs',[]):

**Comment by mikeboers** [comment]
Do you mean [this line][x]?

[x]: https://github.com/mikeboers/PyAV/blob/af9decdf169193d98b2a9297fc9f0437525c5281/setup.py#L80

Why was it failing?

**Comment by ilteroi** [comment]
the error was in line 56 (don't have a clue why i wrote 70):
     for root in extension_extra.get('library_dirs'):

the default return value is None, which isn't iterable. and apparently library_dirs does not exist with my setup.

btw, i've found https://github.com/mikeboers/PyAV/issues/28, which is similar but was closed last year

**Comment by reticulatingspline** [comment]
Similar issue with trying to build from source or use pip to upgrade from 0.2.2 to 0.2.3

Collecting av
  Downloading av-0.2.3.tar.gz (66kB)
    100% |################################| 69kB 2.6MB/s
    Complete output from command python setup.py egg_info:
    Traceback (most recent call last):
      File "<string>", line 20, in <module>
      File "/tmp/pip-build-NxRWz8/av/setup.py", line 184, in <module>
        if check_for_func(libs, func):
      File "/tmp/pip-build-NxRWz8/av/setup.py", line 70, in check_for_func
        lib_path = find_library(lib_name)
      File "/tmp/pip-build-NxRWz8/av/setup.py", line 56, in find_library
        for root in extension_extra.get('library_dirs'):
    TypeError: 'NoneType' object is not iterable

**Comment by mikeboers** [comment]
I made the tiny fix for that, and then added `python setup.py doctor` to dump out the discovered config information. Can you post the results of that?

**Comment by reticulatingspline** [comment]
Got it.

```
~/src/PyAV ±master> sudo apt-get install libavformat-dev                                                                                                      2 ↵
Reading package lists... Done
Building dependency tree
Reading state information... Done
libavformat-dev is already the newest version.
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.

~/src/PyAV ±master » python setup.py doctor
Could not find avformat with ctypes; looked in:
        libavformat.so.54
Could not find avutil with ctypes; looked in:
        libavutil.so.52
Could not find avcodec with ctypes; looked in:
        libavcodec.so.54
Could not find ['avformat', 'avutil', 'avcodec'] with ctypes.util.find_library
Some libraries can not be found for inspection; aborting!
```

**Comment by mikeboers** [comment]
Heh. Yeah it would abort there, wouldn't it,.. :stuck_out_tongue:

**Comment by mikeboers** [comment]
Good news! My brand new Ubuntu 14.04.2 virtual machine is doing exactly the same thing! This should make it a bit easier...

**Comment by mikeboers** [comment]
TL;DR: Try again.

It seems like this version combination of Libav/Ubuntu/pkg-config results in pkg-config not returning the library path (perhaps since it is in the default system path?).

So, now we dump the linker script from `ld` and parse `SEARCH_DIR` from that.

**Comment by mikeboers** [comment]
Do let me know if this works. I'll bundle up a new patch revision for the PyPI, and work on cleaning up the stuff I added in haste later on...

**Comment by mikeboers** [comment]
I'm going to assume this works for you; I've released it as 0.2.4.

**Comment by reticulatingspline** [comment]
My apologies. It does. I just built 0.2.4 this morning. Thanks!
